### DISCUSSION TITLE
Updating docker image from mirror.gcr.io without tags

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#10302] [Request Help] [answered]
I'm in the situation that the Gitlab installation where I have renovate running on lives in the google cloud and is isolated from the outside world. So I cannot access `index.docker.io`.

In the google cloud there is a partial mirror of the docker registry called `mirror.gcr.io` and this mirror does contain the renovate image.
When I list the tags that are available to me it shows
```
$ gcloud container images list-tags mirror.gcr.io/renovate/renovate
DIGEST        TAGS    TIMESTAMP
4e100b2dda71  latest  2021-06-02T22:14:03
5966ec2111e3          2021-06-02T14:26:12
d4b9fb071e8b          2021-06-02T06:52:15
56e43eeb0c22          2021-06-02T01:05:30
8fcda950ee0b          2021-06-01T19:50:44
b41b51c51565          2021-06-01T13:36:29
9ea495908e20          2021-05-31T14:44:52
```

As you can see none of the version tags are included, only `latest`.

For our usecase we take the provided renovate image and extend it with a few tweaks we need for our situation.
So I have a `Dockerfile` with `FROM renovate/renovate:25.33.3` and I created a `renovate.json` with this to enable automatic updating of the renovate image:
```
{
  "enabledManagers": [
    "dockerfile"
  ],
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "registryUrls": ["https://mirror.gcr.io"],
      "labels": ["docker-update"]
    }
  ]
}
```

Now I run renovate on our project (which is called `renovate/renovate-image`) and I get this as part of the output:
```
DEBUG: Found dockerfile package files (repository=renovate/renovate-image)
DEBUG: Found 1 package file(s) (repository=renovate/renovate-image)
 INFO: Dependency extraction complete (repository=renovate/renovate-image)
       "baseBranch": "master",
       "stats": {
         "managers": {"dockerfile": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
DEBUG: getLabels(https://mirror.gcr.io, renovate/renovate, latest) (repository=renovate/renovate-image)
DEBUG: getManifestResponse(https://mirror.gcr.io, renovate/renovate, latest) (repository=renovate/renovate-image)
DEBUG: found labels in manifest (repository=renovate/renovate-image)
       "labels": {
         "maintainer": "Rhys Arkins <rhys@arkins.net>",
         "name": "renovate",
         "org.opencontainers.image.licenses": "AGPL-3.0-only",
         "org.opencontainers.image.source": "https://github.com/renovatebot/renovate",
         "org.opencontainers.image.url": "https://renovatebot.com",
         "org.opencontainers.image.version": "25.36.0"
       }
DEBUG: Found no results from datasource that look like a version (repository=renovate/renovate-image, dependency=renovate/renovate)
       "result": {
         "releases": [],
         "sourceUrl": "https://github.com/renovatebot/renovate",
         "registryUrl": "https://mirror.gcr.io"
       }
```

What I see is that it correctly finds the `latest` image version available and in the manifest it clearly indicates
`"org.opencontainers.image.version": "25.36.0"`
However renovate does not see this as a version identifier: `Found no results from datasource that look like a version`

My expectation is that this label would have been picked up automatically as a valid version identifier.

How do I configure this for my project?

**Comment by rarkins** [comment, marked as answer]
The `labels` are being logged from informational purposes and that's not where/how we extract the version for Docker images. Versions are extracted from the tag list.

**Reply by nielsbasjes** [reply]
Yes, I expected that the tags were the primary way of figuring out the version.
Is there a way to use a label as the version identifier?

**Reply by rarkins** [reply]
Not currently any way to use labels as a version identifier. Essentially you'd get just *one* version if so.

I would also expect that most images other than Renovate would not insert their image version in labels - it's not very common.

**Reply by nielsbasjes** [reply]
Yes I agree. It is very uncommon.
So having renovate look at the "org.opencontainers.image.version" label would only solve my problem for a subset of the images.
Thanks for the help

**Reply by nielsbasjes** [reply]
Note that there is a ticket at google about the root cause of this: https://issuetracker.google.com/issues/137170318
